x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/DataStreamLicenceDowngradeIT.java

```diff
@@ -0,0 +1,489 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.logsdb;
+
+import org.elasticsearch.index.mapper.SourceFieldMapper;
+
+import java.io.IOException;
+import java.util.List;
+
+public class DataStreamLicenceDowngradeIT extends DataStreamLicenseChangeIT {
+    @Override
+    protected void applyInitialLicense() throws IOException {
+        startTrial();
+    }
+
+    @Override
+    protected void licenseChange() throws IOException {
+        startBasic();
+    }
+
+    @Override
+    protected List<TestCase> cases() {
+        return List.of(new TestCase() {
```

Comment:
Nit: I'd add the test cases as separate subclasses below to fascilitate debugging, and include them in the list here.